### PR TITLE
17. As a user I want my lists protected by google and firebase authentication. I want the log in/ sign up to be separated from the rest of the app for clarity and protection/privacy.

### LINKED ISSUE #41: 17. Sign in page and protected routes
## Summary

Create a Login/ Sign up page the allows the user to create an account or log into their account.  Create a gate to keep users lists safe and private/ unaccessible without login.

## Acceptance criteria

- [x] The signup page resembles the figma design and basic theming
- [x] The layout should be responsive based on screen sizes
- [x] Only the sign up page is navigable without permissions. 

## Notes

  *useAuth custom hook
  *private/public routes

### PR DESCRIPTION
**Pull request by StefieCaff**
## Description

This code introduces our branding and guides the user to use their google account to use our app. 

For concise and clear functionality  the login/ sign up button is shared and labeled with google authentication.  

A not found page was also added. In the event the user tries to navigate to a non existing page, they will be navigated here. A message will alert them and a link will allow them to navigate back to the Login page, which in turn will navigate them home if they are authenticated, or allow them to re-authenticate.

I would like to add other forms of authentication in the event a user chooses not to use google.

## Related Issue
closes #41 

## Acceptance Criteria

- [x] The signup page resembles the figma design and basic theming
- [x] The layout should be responsive based on screen sizes
- [x] Only the sign up page is navigable without permissions. 

## Type of Changes

`accessibility` or `enhancement` `security`

### Before

issue-17-before-2

### After

issue-17-after

## Testing Steps / QA Criteria

* git pull from main
* git checkout sc-issue-17
* npm start
* sign out if you are already signed in
* copy and past this into the browser address bar to attempt to access the home page: http://localhost:3000/
* copy and past this into the browser address bar to attempt to access the list page: http://localhost:3000/list
* copy and past this into the browser address bar to access the not found page: http://localhost:3000/kawkaw
* sign up/ log in to the app. Navigate to the home and list page. Sign out of the app. Repeat as many times as desired.
*  inspect the page for screen size compatibility down to 280 px ( Galaxy Fold)

### CODE REVIEW

**Review by 3campos — CHANGES_REQUESTED** [review]
Good job on this! I ran the code locally. Navigating the web pages worked as expected when signed in and not signed in. Nice work on styling the login page.

Two changes that I think should be made (regarding the error page) before merging are:

1. Add a visible link (or alternative) to the error page so that the user can click and return to the home page. You can make the call regarding how to approach this. Some options that come to my mind are: format a piece of the text as a simple visible hyperlink; make a button; or have the page automatically redirect after 2 to 3 seconds.
2. Since this error page will only appear when a user is already signed in, it sounds better if the last sentence of the page ("Try logging back in.") is replaced with something such as: "Click here to be redirected to the home page". If you're using an automatic redirect, then use something like: "Don't worry, we're sending in an extraction team to bring you back home."

One thing I noticed that is not a blocker is that the screen briefly flashes entirely in a dark color before the loading spinner appears. I took a screen recording of this happening. It's a very brief flash and we can create another issue ("bug fix"), if you think it negatively impacts the user experience. Again, this is not a blocker.


https://github.com/the-collab-lab/tcl-66-smart-shopping-list/assets/105468537/07f0f5b6-a6fd-4bb1-aa3c-8514044f7366

**Comment by StefieCaff** [comment]
@3campos thanks for your suggestions 👌🏽
-- let me know what you think of my implementation of the styling and the wording on `Err.jsx`.
-- I was looking into it during my problem solving. I believe that the flash you are seeing is because of how the firebase authentication works. Here are the [docs](https://firebase.google.com/docs/reference/js/auth.md#signinwithredirect) for the firebase function that is implemented. I will look into it more if I can, but we might just have to accept it for now.

I also changed the color of the graph paper. I am open to any more changes, just add them to a review and I will work on it.

**Review by 3campos — APPROVED** [review]
The darker graph paper lines contrast better with our light background and you made good changes to the error page. Merge when you're ready!

File: src/views/Err.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+					</h1>
+				</div>
+				<div className="w-full max-w-xl p-7 lg:p-10 gap-5 rounded-lg shadow-lg bg-white">
+					<h2 className="leading-6 font-semibold text-lg">
+						Eeep, it seems the page you're looking for doesn't exist.
+					</h2>
+					<div className="inline-block border-t-1 self-stretch border-loginLine/50"></div>
+					<div className="px-4">
+						<h3>
+							Please navigate back to the{' '}
```
Comment (nick-zanetti) [inline src/views/Err.jsx]: Maybe this could just say "Click here to navigate back to your lists"?

**Review by nick-zanetti — APPROVED** [review]
This looks good to me! One thought: when users click the "verify with google" button, is there a way they can choose a specific google account to use? Does this automatically use a google account if you are logged in to google with it on the browser?

File: src/views/Err.jsx (hunk 1 above)
Comment (StefieCaff) [reply src/views/Err.jsx]: yeah, that sounds better for sure, thanks!

**Comment by StefieCaff** [comment]
@nick-zanetti this is a good question, I will run a few tests in browsers that are not associated with my already authenticated account. I have seen when Aloe logs in, they have several accounts active in their browser and it allows then to click the one they want to use.

**Comment by eonflower** [comment]
> This looks good to me! One thought: when users click the "verify with google" button, is there a way they can choose a specific google account to use? Does this automatically use a google account if you are logged in to google with it on the browser?

As Stefie said, mine allows me to choose which google account, but I think it is based on what accounts you already have signed in on your computer. All the sign in logic was already built in the project before we started working on it.

**Comment by nick-zanetti** [comment]
I see! Think you can disregard my suggestion then @StefieCaff

**Comment by StefieCaff** [comment]
> I see! Think you can disregard my suggestion then @StefieCaff

Led to a good update actually!! Thanks @nick-zanetti and @eonflower

**Pull request merged**